workspaces/src/network/result.rs

```diff
@@ -133,3 +205,84 @@ impl From<CallResult> for ViewResultDetails {
         }
     }
 }
+
+/// The execution outcome of a transaction. This type contains all data relevant to
+/// calling into a function, and getting the results back.
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[non_exhaustive]
+pub struct ExecutionOutcome {
+    pub block_hash: CryptoHash,
+    /// Logs from this transaction or receipt.
+    pub logs: Vec<String>,
+    /// Receipt IDs generated by this transaction or receipt.
+    pub receipt_ids: Vec<CryptoHash>,
+    /// The amount of the gas burnt by the given transaction or receipt.
+    pub gas_burnt: Gas,
+    /// The id of the account on which the execution happens. For transaction this is signer_id,
+    /// for receipt this is receiver_id.
+    pub executor_id: AccountId,
+    /// Execution status. Contains the result in case of successful execution.
+    pub(crate) status: ExecutionStatusView,
+}
+
+impl ExecutionOutcome {
+    /// Checks whether this execution outcome was a success. Returns true if a success value or
+    /// receipt id is present.
+    pub fn is_success(&self) -> bool {
+        matches!(
+            self.status,
+            ExecutionStatusView::SuccessValue(_) | ExecutionStatusView::SuccessReceiptId(_)
+        )
+    }
+
+    /// Checks whether this execution outcome was a failure. Returns true if it failed with
+    /// an error or the execution state was unknown or pending.
+    pub fn is_failure(&self) -> bool {
+        matches!(
+            self.status,
+            ExecutionStatusView::Failure(_) | ExecutionStatusView::Unknown
+        )
+    }
+
+    /// Converts this [`ExecutionOutcome`] into a Result type, where the failure is converted
+    /// to an [`anyhow::Error`] object which can be downcasted later.
+    pub fn into_result(self) -> anyhow::Result<ValueOrReceiptId> {
+        match self.status {
+            ExecutionStatusView::SuccessValue(value) => Ok(ValueOrReceiptId::Value(value)),
+            ExecutionStatusView::SuccessReceiptId(hash) => {
+                Ok(ValueOrReceiptId::ReceiptId(CryptoHash(hash.0)))
+            }
+            ExecutionStatusView::Failure(err) => {
+                Err(anyhow::anyhow!("Execution failed: {:?}", err))
+            }
```

Comment:
Will need to directly change this formatted anyhow error in the future when https://github.com/near/nearcore/pull/6367 makes it into a near_primitives release